code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt

```diff
@@ -0,0 +1,124 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.templates
+
+import android.app.NotificationManager
+import com.adobe.marketing.mobile.notificationbuilder.internal.NotificationPriority
+import com.adobe.marketing.mobile.notificationbuilder.internal.NotificationVisibility
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateType
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.MapData
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+@RunWith(MockitoJUnitRunner::class)
+class AEPPushTemplateTest {
+    @Test
+    fun `Test BasicPushTemplate initialization with Map`() {
+        val aepPushTemplate = provideMockedBasicPushTemplateWithRequiredData()
+        assertEquals(MOCKED_TITLE, aepPushTemplate.title)
+        assertEquals(MOCKED_BODY, aepPushTemplate.body)
+        assertEquals(MOCKED_PAYLOAD_VERSION, aepPushTemplate.payloadVersion)
+    }
+
+    @Test
+    fun `Test exception with missing data adb_title`() {
+        val aepPushBasicData = MockAEPPushTemplateDataProvider.getMockedDataMapWithRequiredData()
+        aepPushBasicData.remove(PushTemplateConstants.PushPayloadKeys.TITLE)
+        val exception = assertFailsWith<IllegalArgumentException> {
+            BasicPushTemplate(MapData(aepPushBasicData))
```

Comment:
Name of the file says its tests for `AEPPushTemplate` but we're creating `BasicPushTemplate` here. Is it supposed to be moved to a different file or is the class under test wrong?

Reply:
i think its using a basic push template as an AEPPushTemplate can't be created directly as it's sealed. i think this is ok as the verification is checking for AEPPushTemplate KVP.

Reply:
Ah ok, that makes sense. Thank you!